comp/autoscaling/workload/def/component.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package workload implements the autoscaling controller
+package workload
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
```

Comment:
Could we try not having as part of the public interface of the component a hard dependency on a third party library? 

Could we create a custom interface that matches what we want to extract from `*corev1.Pod`? 

That way this component is more reusable in the future and importing the interface does not bring `"k8s.io/api/core/v1"`